service/src/test/java/org/broadinstitute/listener/relay/inspectors/SamResourceClientTest.java

```diff
@@ -0,0 +1,57 @@
+package org.broadinstitute.listener.relay.inspectors;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.Map;
+import java.util.Optional;
+import org.broadinstitute.dsde.workbench.client.sam.ApiClient;
+import org.broadinstitute.dsde.workbench.client.sam.ApiException;
+import org.broadinstitute.dsde.workbench.client.sam.ApiResponse;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class SamResourceClientTest {
+
+  private SamResourceClient samResourceClient;
+
+  @Mock private TokenChecker tokenChecker;
+  @Mock private ApiClient apiClient;
+
+  @BeforeEach
+  void setUp() throws IOException, InterruptedException, ApiException {
+    samResourceClient = new SamResourceClient("resourceId", apiClient, tokenChecker);
+  }
+
+  @Test
+  void checkWritePermission_sucess() throws IOException, InterruptedException, ApiException {
+    var expiresAt = Instant.now().plusSeconds(100);
+    var oauthResponse = new OauthInfo(Optional.of(expiresAt), "");
+    when(tokenChecker.getOauthInfo(any())).thenReturn(oauthResponse);
+    var apiResponse = new ApiResponse(200, Map.of(), true);
```

Comment:
Using a parameterized type like `ApiResponse` without a type argument will generate a warning (and probably sonar will complain too). Looking at this code, `Object` should work:
```suggestion
    var apiResponse = new ApiResponse<Object>(200, Map.of(), true);
```